src/CommunityToolkit.Aspire.Hosting.RavenDB/RavenDBBuilderExtensions.cs

```diff
@@ -0,0 +1,219 @@
+        databaseName ??= name;
+
+        builder.Resource.AddDatabase(name, databaseName);
+        var databaseResource = new RavenDBDatabaseResource(name, databaseName, builder.Resource);
+
+        string? connectionString = null;
+
+        builder.ApplicationBuilder.Eventing.Subscribe<ConnectionStringAvailableEvent>(databaseResource, async (@event, ct) =>
+        {
+            connectionString = await databaseResource.ConnectionStringExpression.GetValueAsync(ct).ConfigureAwait(false);
+
+            if (connectionString == null)
+                throw new DistributedApplicationException($"ConnectionStringAvailableEvent was published for the '{databaseResource.Name}' resource but the connection string was null.");
+        });
+
+        var healthCheckKey = $"{name}_check";
+        builder.ApplicationBuilder.Services.AddHealthChecks()
+            .AddRavenDB(sp => connectionString ?? throw new InvalidOperationException("Connection string is unavailable"),
+                databaseName: databaseName,
+                name: healthCheckKey);
+
+        return builder.ApplicationBuilder
+            .AddResource(databaseResource);
+    }
+
+    /// <summary>
+    /// Adds a bind mount for the data folder to a RavenDB container resource.
+    /// </summary>
+    /// <param name="builder">The resource builder for the RavenDB server.</param>
+    /// <param name="source">The source directory on the host to mount into the container.</param>
+    /// <param name="isReadOnly">Indicates whether the bind mount should be read-only. Defaults to false.</param>
+    /// <returns>The <see cref="IResourceBuilder{T}"/> for the RavenDB server resource.</returns>
+    public static IResourceBuilder<RavenDBServerResource> WithDataBindMount(this IResourceBuilder<RavenDBServerResource> builder, string source, bool isReadOnly = false)
+    {
+        ArgumentNullException.ThrowIfNull(builder);
+        ArgumentNullException.ThrowIfNull(source);
+
+        return builder.WithBindMount(source, "/var/lib/ravendb/data", isReadOnly);
+    }
+
+    /// <summary>
+    /// Adds a named volume for the data folder to a RavenDB container resource.
+    /// </summary>
+    /// <param name="builder">The resource builder for the RavenDB server.</param>
+    /// <param name="name">Optional name for the volume. Defaults to a generated name if not provided.</param>
+    /// <param name="isReadOnly">Indicates whether the volume should be read-only. Defaults to false.</param>
+    /// <returns>The <see cref="IResourceBuilder{T}"/> for the RavenDB server resource.</returns>
+    public static IResourceBuilder<RavenDBServerResource> WithDataVolume(this IResourceBuilder<RavenDBServerResource> builder, string? name = null, bool isReadOnly = false)
+    {
+        ArgumentNullException.ThrowIfNull(builder);
+
+#pragma warning disable CTASPIRE001
+        return builder.WithVolume(name ?? VolumeNameGenerator.CreateVolumeName(builder, "data"), "/var/lib/ravendb/data", isReadOnly);
```

Comment:
I see that both "WithDataBindMount()" and "WithDataVolume()" mount into same directory, this means they should not be used together?

Reply:
Right! they cannot be used together because they mount into the same directory, which would cause a conflict